sfa_dash/__init__.py

```diff
@@ -18,21 +19,28 @@ def create_app(config=None):
     register_jinja_filters(app)
     error_handlers.register_handlers(app)
 
-    session_storage = SessionStorage()
-    auth0_bp = make_auth0_blueprint(
+    if app.config['SQLALCHEMY_DATABASE_URI']:
+        db.init_app(app)
+        db.create_all(app=app)
+
+    make_auth0_blueprint(
+        app,
         base_url=app.config['AUTH0_OAUTH_BASE_URL'],
         storage=session_storage)
-    app.register_blueprint(auth0_bp, url_prefix='/login')
-    app.route('/logout')(logout)
 
     def protect_endpoint():
-        if not oauth_request_session.authorized:
-            # means we have a token, not necessarily that it
-            # hasn't expired, but refreshing is handled
-            # by request_oauthlib and oauthlib
-            # and the api validates expiration
-            session['redirect_path'] = request.path
-            return redirect(url_for('auth0.login'))
+        try:
+            authorized = oauth_request_session.authorized
+        except ValueError:
+            return redirect(url_for('index'))
+        else:
+            if not authorized:
+                # means we have a token, not necessarily that it
+                # hasn't expired, but refreshing is handled
+                # by request_oauthlib and oauthlib
+                # and the api validates expiration
+                session['redirect_path'] = request.path
+                return redirect(url_for('auth0.login'))
```

Comment:
To retain the redirect after login, I think the except block needs to follow the same logic as the else here, otherwise it only redirects users who haven't been logged in recently.